docs/zh_CN/2023-04-12-那么，接下来是什么.md

```diff
@@ -0,0 +1,749 @@
+
+<p align="center"><img alt="lighthouse" width="720" src="https://user-images.githubusercontent.com/2213682/148652288-bd6e452a-f6ba-417d-8972-9d98d2f715a4.png" /></p>
+
+Lighthouse检测出所有资源仅约200KB，0.56秒。但该网站包含大约几兆字节的polyfill。Lighthouse只检测出不到一半的应检测内容[^half]，但即把它能检测的都检测出来，它也只是所有加载的polyfill的一小部分。其余的在哪里？现代浏览器真的需要它们吗？问题是，本地模块支持的下限太低——在这种情况下，“现代”浏览器仍将需要旧版IE所需的大多数polyfill，以提供稳定的JS功能。因此，部分polyfill显示在“未使用JavaScript”部分中，需要6.41s，另外的部分根本没有显示......
+
+[^half]: https://github.com/GoogleChrome/lighthouse/issues/13440
+
+从我开始写core-js的一开始，我就一直在考虑创建一个Web服务，为浏览器提供所需的polyfill服务。
+
+未提供此类服务是core-js落后于另一个项目的唯一方面。英国《金融时报》的polyfill-service[^serv]就基于这一概念，这是一个很牛的服务。这个项目的主要问题是，服务很牛，但是用了糟糕的polyfill。这个项目只是polyfill了core-js提供的ECMAScript功能的一小部分，大多数polyfill是第三方的，他们并不是为一起工作而设计的，太多polyfill没有正确遵循规范，或者太粗糙，或者使用起来很危险（例如，WeakMap看样子是在一步步实现规范[^step]，但缺乏一些特别手段，使其导致内存泄漏和线性访问时间，这就不太好了，还有更多其他问题——它不是在IE11这类引擎中修补、修复和重用本地实现，而是不接受可迭代参数，这使得WeakMap将会被完全替换[^replace]）。一些优秀的开发人员时不时尝试修复这个问题，但是polyfill本身拥有的时间实在是太少了，因此它还远远达不到可被推荐的地步。
+
+
+[^serv]: https://polyfill.io/
+[^step]: https://github.com/Financial-Times/polyfill-library/blob/554248173eae7554ef0a7776549d2901f02a7d51/polyfill/WeakMap/polyfill.js
+[^replace]: https://github.com/Financial-Times/polyfill-library/blob/554248173eae7554ef0a7776549d2901f02a7d51/polyfill/WeakMap/detect.js
+
+以适当的形式创建这样的服务需要制作和维护许多新组件。我独自一人从事core-js开发，项目没有得到任何公司的必要支持，开发纯粹是建立在热情之上，我需要资金来养活自己和家人，所以我没有时间和其他资源去开发它。然而，在其他任务范围内，我已经制作了一些必需的组件，与一些用户的讨论让我相信，创建一个最大限度简化的服务，然后在你自己的服务器上运行它就足够了。
+
+我们已经拥有了最好的polyfill集、适当的兼容性数据以及可以为目标浏览器创建捆绑包的构建器。前面提到的@core-js/collector可以用来做优化——仅获取所需的模块子集、转译器/捆绑器的插件——删除不必要的polyfill。仅仅是缺少一个用于用户代理规范化的工具和一个将这些组件绑定在一起的服务，让我们称它为@core-js/service。
+
+#### 在一个完美的世界里，它应该是什么样子？
+
+- 捆绑你的项目。捆绑器端的插件会从你的依赖项中删除所有polyfill导入（包括第三方的，没有全局污染的）。你的捆绑包将不包含任何polyfill。
+- 运行@core-js/service。当你运行它时，@core-js/collector会检查你的所有前端代码库、所有入口点，包括依赖项，并收集所有必需的polyfill的列表。
+- 用户加载一个页面，并向服务请求polyfill捆绑包。服务为客户端提供了一个为目标浏览器编译的捆绑包，其中包含所需的polyfill子集并使用所允许的语法。
+
+这样，使用这种工具，现代浏览器如果自身够用，将根本不加载polyfill，旧浏览器将只加载所需的优化后的polyfill。
+
+---
+
+以上大部分内容都是关于如何将发送给客户端的polyfill最小化——这些只是core-js范围内想要实现东西的一小部分，如果你明白做到这一点需要大量的工作，而且它可以显著改善Web开发，那就足够了。它是否会真的被做出来，以及它是FOSS的还是商业的，取决于你。
+
+
+
+# 结论
+
+这是我最后一次尝试将core-js保留为具有适当质量和功能水平的免费开源项目。这是最后一次传递信息：在开源的另一边有真实的人，他有家庭需要养活，有问题需要解决。
+
+如果你或你的公司以这样或那样的方式使用core-js，并且对你的供应链质量感兴趣，请支持本项目：
+
+- Open Collective[^open]
+- Patreon[^pat]
+- Boosty[^boosty]
+- Bitcoin （bc1qlea7544qtsmj2rayg0lthvza9fau63ux0fstcz）
+- 支付宝[^alipay] 
+
+
+[^open]: https://opencollective.com/core-js
+[^pat]: https://patreon.com/zloirock
+[^boosty]: https://boosty.to/zloirock
+[^alipay]: https://user-images.githubusercontent.com/2213682/219464783-c17ad329-17ce-4795-82a7-f609493345ed.png
+
+**如果你能在Web标准和开源方面提供一份好工作，请联系我。**
+
+请随时在此issue帖子中添加评论:
+
+https://github.com/zloirock/core-js/issues/1179
+
+Denis Pushkarev[^denis]，2023年2月14日
+
+
+[^denis]: https://github.com/zloirock
+
+
+##### 翻译｜卫剑钒
+##### 译者注：这篇翻译一并发布在我的公众号“卫sir说”上:[《一个开源人的心酸哭诉》](https://mp.weixin.qq.com/s/EmxfaSOtqNAuC7XmzlHa4w)
```

Comment:
Well, I don't think adding links for yourself is appropriate.

Reply:
OK, I have removed the link.